### PR TITLE
Feat/k8s

### PR DESCRIPTION
**Pull request by corpulent**

### CODE REVIEW

**Comment by corpulent** [comment]
@itssamuelrowe test against https://staging-api.ctk.dev/v1/

File: services/frontend/src/services/generate.ts [hunk 1]
```diff
@@ -1,7 +1,16 @@
 import { API_SERVER_URL } from "../constants";
 
-export const generateHttp = (data: string) => {
-  return fetch(`${API_SERVER_URL}/generate/`, {
+export const generateHttp = (data: string, manifest: number) => {
+  let endpoint = `${API_SERVER_URL}/generate/`;
+  if (manifest === 0) {
```
Comment (itssamuelrowe) [inline services/frontend/src/services/generate.ts:5]: Why are you using number literals to indicate the type of the manifest? It'd be better if you could create constants for them. `manifestTypes.DOCKER_COMPOSE`.

File: services/frontend/src/components/Project/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+const ManifestSelect = (props: IManifestSelectProps) => {
+  const { onSave, isLoading, projectData, isAuthenticated } = props;
+  const [visibility, setVisibility] = useState(false);
+  const [projectName, setProjectName] = useState("Untitled");
+
+  const handleNameChange = (e: any) => {
+    setProjectName(e.target.value);
+  };
```
[the remark this answers is not in the document]
Comment (corpulent) [reply services/frontend/src/components/Project/Header.tsx:20]: Like this?

```
  const handleNameChange = useCallback((e: any) => {
    setProjectName(e.target.value);
  }, []);
```

File: services/frontend/src/components/Project/CodeBox.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          setGeneratedCode(data["code"]);
+        } else {
+          setGeneratedCode("");
+        }
+
+        if (data["error"].length) {
+          setGeneratedCode("");
+          toaster(`error ${data["error"]}`, "error");
+        }
+      })
```
[the remark this answers is not in the document]
Comment (corpulent) [reply services/frontend/src/components/Project/CodeBox.tsx]: Some refactoring needs to happen, so lets make an issue.

**Pull request merged**